static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js

```diff
@@ -0,0 +1,38 @@
+define([
+    'bean',
+    'bonzo',
+    'fastdom',
+    'common/utils/$'
+], function (
+    bean,
+    bonzo,
+    fastdom,
+    $
+) {
+
+
+    function init() {
+        var $scrollEl = $('.hosted-gallery__scroll-container');
+        var $images = $('.hosted-gallery__image');
+        var noImages = $images.length;
+        if (!$scrollEl.length) {
+            return;
+        }
+
+        bean.on($scrollEl[0], 'scroll', function (e) {
```

Comment:
You'd really want to `throttle` this.
You could use the [`window:throttledScroll`](https://github.com/guardian/frontend/blob/6af3c99aeee98156685296b817cb3ec1599e8bbb/static/src/javascripts/bootstraps/standard/main.js#L197) event too.